TerrainSystemComponent: Add RPISytem as a dependency

msat-huawei opened the pull request:
## What does this PR do?

The TerrainSystemComponent uses the `PassSystemInterface` in its Activate function, but does not depend on the RPISystem.
This PR adds this needed dependency.

## How was this PR tested?

Tested on windows with the startergame-assets project

nick-l-o3de commented:
I'm guessing what happened in the tests is that previously, the test only exercised parts of Terrain that did not actually rely on the RPISystem, so the component was able to activate without it present.

Now, the Terrain Component requires that the RPI System component is present in order to activate, and thus the test fails.

msat-huawei commented:
I added a mock implementation of the RPISystemComponent for the Terrain Tests since this seemed to be the proper way to handle this. I also moved the RPISystem and ImageSystem into this system component.
